Source/DafnyCore/AST/Members/Method.cs

```diff
@@ -270,6 +283,29 @@ public class Method : MemberDecl, TypeParameter.ParentType,
         resolver.ConstrainTypeExprBool(e.E, "Precondition must be a boolean (got {0})");
       }
 
+      // TODO: May not be the right place to set the default, and may want something similar to InferredDecreases
+      // TODO: Might be clearer to add a resolved `reads *` after instead,
+      // or to add it unresolved as we do here but in an earlier hook like
+      // ResolveNamesAndInferTypesForOneDeclarationInitial?
+      var readsClausesOnMethodsEnabled = resolver.Options.Get(CommonOptionBag.ReadsClausesOnMethods);
+      if (readsClausesOnMethodsEnabled && !Reads.Any() && !IsLemmaLike) {
+        // Note that `reads *` is the right default for backwards-compatibility,
+        // but we may want to infer a sensible default like decreases clauses instead in the future.
+        Reads.Add(new FrameExpression(tok, new WildcardExpr(tok), null));
+      }
+      foreach (FrameExpression fe in Reads) {
+        resolver.ResolveFrameExpressionTopLevel(fe, FrameExpressionUse.Reads, this);
+        if (IsLemmaLike) {
+          resolver.reporter.Error(MessageSource.Resolver, fe.tok, "{0}s are not allowed to have reads clauses",
```

Comment:
```suggestion
          resolver.reporter.Error(MessageSource.Resolver, fe.tok, "{0}s do not have reads clauses but they can read anything they need",
```

Reply:
Good idea, although I'm going with "(they are allowed to read all memory locations)" instead.